repo/format/format_manager_test.go

```diff
@@ -169,13 +171,148 @@ func TestInitialize(t *testing.T) {
 		format.ErrAlreadyInitialized)
 }
 
+func TestInitializeWithRetention(t *testing.T) {
+	ctx := testlogging.Context(t)
+
+	startTime := time.Date(2020, 1, 1, 12, 0, 0, 0, time.UTC)
+	ta := faketime.NewTimeAdvance(startTime, 0)
+	nowFunc := ta.NowFunc()
+
+	st := blobtesting.NewVersionedMapStorage(nowFunc).(cache.Storage)
+	blobCache := format.NewMemoryBlobCache(nowFunc)
+	mode := blob.Governance
+	period := time.Hour * 48
+
+	// success
+	require.NoError(t, format.Initialize(
+		ctx,
+		st,
+		&format.KopiaRepositoryJSON{},
+		rc,
+		format.BlobStorageConfiguration{
+			RetentionMode:   mode,
+			RetentionPeriod: period,
+		},
+		"some-password",
+	))
+
+	mgr, err := format.NewManagerWithCache(ctx, st, cacheDuration, "some-password", nowFunc, blobCache)
+	require.NoError(t, err, "getting format manager")
+
+	// New retention parameters should be available from the format manager.
+	blobCfg := mustGetBlobStorageConfiguration(t, mgr)
+	assert.Equal(t, mode, blobCfg.RetentionMode)
+	assert.Equal(t, period, blobCfg.RetentionPeriod)
+
+	// Attempting to touch the blobs the format manager writes should return
+	// errors as they should have retention enabled. Mod time adjustment (duration
+	// param) doesn't matter in this context.
+	_, err = st.TouchBlob(ctx, format.KopiaRepositoryBlobID, time.Minute)
+	assert.ErrorIs(t, err, blobtesting.ErrBlobLocked, "Altering locked repo blob should fail")
+
+	_, err = st.TouchBlob(ctx, format.KopiaBlobCfgBlobID, time.Minute)
```

Comment:
Also, It'd be desired to verify that the initial format blob PUT was set with the desired `RetainUntilDate` and `Mode`.

Reply:
I can indirectly check the retain until date by advancing the clock, similar to the [tests](https://github.com/kopia/kopia/blob/22cba7003d789b71a9512a66e52781eeb4346033/repo/maintenance/blob_retain_test.go#L59) for retention extension. There doesn't seem to be a way to directly check those properties though. I'm going to save the larger change of checking mode and retain until for another PR but will advance the clock and try `TouchBlob` again in this PR if that's alright with you

Reply:
Can we access the object-locking map underneath and add a method to it to expose full metadata ?

Reply:
yea, something like that should be doable, though it will require a slightly bigger change to store retention mode in the blobtesting implementation. Can I do that in another PR so this one doesn't get too big? If we're going to do that we should probably make similar checks in other retention-related tests

Reply:
These tests are (more than) sufficient to verify the fix introduced in the PR

Reply:
> it will require a slightly bigger change to store retention mode in the blobtesting implementation

Makes sense. Ok to take this in without the metadata evaluation. The blobtesting can probably be improved going forward:

>  If we're going to do that we should probably make similar checks in other retention-related tests

Yes. Separate PR.

https://github.com/kopia/kopia/issues/3218